src/cluster/aks.go

```diff
@@ -290,132 +282,6 @@ func makeInfrastructureResourceGroupName(resourceGroupName, clusterName, locatio
 	return fmt.Sprintf("MC_%s_%s_%s", resourceGroupName, clusterName, location)
 }
 
-func makeResourceGroupScope(subscriptionID, resourceGroupName string) string {
-	return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", subscriptionID, resourceGroupName)
-}
-
-func (c *AKSCluster) assignStorageAccountContributorRole() error {
-	c.log.Infof("Assign %s role to all VMs in %s cluster", StorageAccountContributor, c.GetName())
-
-	cc, err := c.getCloudConnection()
-	if err != nil {
-		return errors.WrapIf(err, "failed to get cloud connection")
-	}
-
-	irgName := c.getInfrastructureResourceGroupName()
-	c.log.Infof("Checking infrastructure resource group [%s] existence", irgName)
-	resp, err := cc.GetGroupsClient().CheckExistence(context.TODO(), irgName)
-	if err != nil {
-		return errors.WrapIf(err, "failed to check resource group existence")
-	}
-	if resp.StatusCode == http.StatusNotFound {
-		return errors.WrapIf(ErrNoInfrastructureRG, "resource group not found")
-	}
-
-	scope := makeResourceGroupScope(cc.GetSubscriptionID(), irgName)
-	c.log.Debugf("Resource group scope: %s", scope)
-
-	c.log.Infof("Search for %s role", StorageAccountContributor)
-	role, err := cc.GetRoleDefinitionsClient().FindByRoleName(context.TODO(), scope, StorageAccountContributor)
-	if err != nil {
-		return errors.WrapIf(err, "failed to assign storage account contributor role to cluster nodes")
-	}
-	if role == nil {
-		return fmt.Errorf("no role found with the given name[%s]", StorageAccountContributor)
-	}
-	c.log.Debugf("Role ID: %s", *role.ID)
-
-	vmClient := cc.GetVirtualMachinesClient()
-
-	c.log.Infof("List virtual machines in resource group [%s]", irgName)
-	virtualMachines, err := vmClient.ListAll(context.TODO(), irgName)
-	if err != nil {
-		return errors.WrapIf(err, "failed to list virtual machines in resource group")
-	}
-
-	if err := enableVMsManagedServiceIdentity(c.log, vmClient, irgName, virtualMachines); err != nil {
-		return errors.WrapIf(err, "failed to enable managed service identity for VMs")
-	}
-
-	if err := assignRoleToVMs(c.log, cc.GetRoleAssignmentsClient(), virtualMachines, role, scope); err != nil {
-		return errors.WrapIf(err, fmt.Sprintf("failed to assign %s role to VMs", StorageAccountContributor))
-	}
-
-	c.log.Info("Role assign succeeded")
-
-	return nil
-}
-
-func enableVMManagedServiceIdentity(log logrus.FieldLogger, client *pkgAzure.VirtualMachinesClient, resourceGroupName string, vm *compute.VirtualMachine) (*compute.VirtualMachine, error) {
```

Comment:
We may want to keep the assignment of identity to VMs as that allows us granting access to cloud resources by adding roles to the identity. An improvement could be done by assigning the identity to the VM at creation time (the same way as we do for VMSSes in for PKE)